posts/2021-02-19-mpcontextpropagation-requesttiming-21003-beta.adoc

```diff
@@ -0,0 +1,314 @@
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= RequestTiming 1.0, MicroProfile Context Propagation 1.2, enhancments to App Manager configuration and Liberty connection management in Open Liberty 21.0.0.3-beta
+Jakub Pomykala <https://github.com/jakub-pomykala>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+
+Open Liberty 21.0.0.3-beta contains four All Beta features (new `expandLocation` property for App Manager configuration, Ability to control whether thread dumps are collected when a hung request is detected in the Request Timing feature (RequestTiming 1.0), MicroProfile Context Propagation 1.2 and Automatic Cleanup of Leaked Connections) and one feature in Jakarta EE 9 (JakartaMail). Also in 21.0.0.3-beta, the link:https://github.com/OpenLiberty/open-liberty/issues/15649[issue in MicroProfile Fault Tolerance 3.0] from 21.0.0.2-beta has now been resolved meaning that the `mpFaultTolerance-3.0` feature and the `microProfile-4.0` convenience feature can be used in the Open Liberty 21.0.0.3-beta.
+
+We have two beta packages for link:{url-about}[Open Liberty]:
+
+* <<allbeta, All Beta Features>>: a larger package that contains all Open Liberty beta features (including Jakarta EE 9 beta features) and GA features and functions.
+* <<jakarta, Jakarta EE 9 Beta Features>>: a lightweight package that contains only the Jakarta EE 9 features.
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools.
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes the following beta features:
+
+* <<expandLocation, expandLocation property for App Manager configuration>>
+* <<requestTiming, Ability to control whether thread dumps are collected when a hung request is detected in the Request Timing feature>>
+* <<MPContext, MicroProfile Context Propagation 1.2>>
+* <<leakedConnections, Automatic Cleanup of Leaked Connections>>
+
+
+These features also join MicroProfile 4.0 features which can be found link:https://openliberty.io/blog/2021/01/26/ee9-messaging-security-21002-beta.html#allbeta[here].
+
+[#expandLocation]
+=== expandLocation property for App Manager configuration
+
+This enhancement allows the user to specify a expansion location (`expandLocation`) on the `applicationManager` configuration to be utilized when the `autoExpand` attribute is set to "true". As currently implemented, when an application is autoExpanded the default location for the expanded files are hard coded to `${server.config.dir}/apps/expanded/`.
+
+The user is now able to configure that location to a new value on the file system. 
+
+Thus for example:
```

Comment:
Remove the word "Thus" ... "For example" is fine

Reply:
Done